### ISSUE TITLE
Support for fractional shares

### ISSUE THREAD

**Issue by nicdumz** [wichtounet/budgetwarrior#29]
I see that  `asset_share.shares` is an `int64`. While this generally makes sense intuitively, this means that it becomes hard to use budgetwarrior for share vesting schemes.

In my case, my employer awards me for instance 20 shares of restricted stock, but after withholding taxes, my broker credits me 18.84 shares. So technically if I wanted to automatically track share awards, I would want to to log the acquisition of 18.84 shares, heh. 

(And 40 becomes.... 37.681 ? for fascinating rounding reasons I suppose)

For curiosity, I wondered if there's an industry standard on what makes sense, but [alas](https://quant.stackexchange.com/questions/60542/are-there-any-standards-for-the-precision-of-stocks-prices-amount-of-stocks-etc) I'm not sure there is anything out there. Fascinating :-)

**Comment by wichtounet** [comment]
Indeed, I have never traded in fractional shares myself, so never saw the need for it on the tracker. But that's a very fair point. 

It should be fairly straightforward to implement with two decimal places since I already have code for that, but looking at your link and your example, I think we will need more decimal places. 

Once I finish the current refactorings I have started, I will try to add this support 👍

**Comment by Scrappy440** [comment]
Is this issue resolved? I would like to work on it
